ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala

```diff
@@ -0,0 +1,897 @@
+      commandId: domain.CommandId,
+      submitters: List[Party],
+  )(implicit conn: Connection): Unit = {
+    val key = deduplicationKey(commandId, submitters)
+    SQL_DELETE_COMMAND
+      .on("deduplicationKey" -> key)
+      .execute()
+    ()
+  }
+
+  override def stopDeduplicatingCommand(
+      commandId: domain.CommandId,
+      submitters: List[Party],
+  )(implicit loggingContext: LoggingContext): Future[Unit] =
+    dbDispatcher.executeSql(metrics.daml.index.db.stopDeduplicatingCommandDbMetrics) {
+      implicit conn =>
+        stopDeduplicatingCommandSync(commandId, submitters)
+    }
+
+  private val SQL_UPDATE_MOST_RECENT_PRUNING = SQL("""
+      |update parameters set participant_pruned_up_to_inclusive={pruned_up_to_inclusive}
+      |where participant_pruned_up_to_inclusive < {pruned_up_to_inclusive} or participant_pruned_up_to_inclusive is null
+      |""".stripMargin)
+
+  private def updateMostRecentPruning(
+      prunedUpToInclusive: Offset
+  )(implicit conn: Connection): Unit = {
+    SQL_UPDATE_MOST_RECENT_PRUNING
+      .on("pruned_up_to_inclusive" -> prunedUpToInclusive)
+      .execute()
+    ()
+  }
+
+  override def prune(
+      pruneUpToInclusive: Offset
+  )(implicit loggingContext: LoggingContext): Future[Unit] =
+    dbDispatcher.executeSql(metrics.daml.index.db.pruneDbMetrics) { implicit conn =>
+      EventsTableDelete.prepareEventsDelete(pruneUpToInclusive).execute()
+      prepareCompletionsDelete(pruneUpToInclusive).execute()
+      updateMostRecentPruning(pruneUpToInclusive)
+      logger.info(s"Pruned ledger api server index db up to ${pruneUpToInclusive.toHexString}")
+    }
+
+  override def reset()(implicit loggingContext: LoggingContext): Future[Unit] =
+    dbDispatcher.executeSql(metrics.daml.index.db.truncateAllTables) { implicit conn =>
+      val _ = SQL(queries.SQL_TRUNCATE_TABLES).execute()
+    }
+
+  private val translation: LfValueTranslation =
+    new LfValueTranslation(
+      cache = lfValueTranslationCache,
+      metrics = metrics,
+      enricherO = enricher,
+      loadPackage = (packageId, loggingContext) => this.getLfArchive(packageId)(loggingContext),
+    )
+
+  override val transactionsReader: TransactionsReader =
+    new TransactionsReader(dbDispatcher, dbType, eventsPageSize, metrics, translation)(
+      servicesExecutionContext
+    )
+
+  private val contractsReader: ContractsReader =
+    ContractsReader(dbDispatcher, dbType, metrics, lfValueTranslationCache)(
+      servicesExecutionContext
+    )
+
+  override val completions: CommandCompletionsReader =
+    new CommandCompletionsReader(dbDispatcher, dbType, metrics, servicesExecutionContext)
+
+  private val postCommitValidation =
+    if (performPostCommitValidation)
+      new PostCommitValidation.BackedBy(contractsReader.committedContracts, validatePartyAllocation)
+    else
+      PostCommitValidation.Skip
+
+  // TODO append-only: cleanup
+  override def storeTransactionState(
```

Comment:
For ease of analysis, I would move this method to where the original was located.